### PR TITLE
Feat(eos_cli_config_gen): Add support for sub-route-map and continue in route-maps

### PR DESCRIPTION
**Pull request by emilarista**
## Change Summary

Adds knobs for sub-route-map and continue in route maps.

## Component(s) name

`arista.avd.eos_cli_config_gen`

## Proposed changes

```yaml
route_maps:
  <rm name>:
    sequence_numbers:
      <seq>:
        ...
        sub_route_map: <sub-RM>
        continue:
          enabled: < true | false >
          sequence_number: < integer >
```

## How to test
Tested with molecule.

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/route-maps.j2 [hunk 1]
```diff
@@ -7,17 +7,17 @@
 {%     for route_map in route_maps | arista.avd.convert_dicts('name') | arista.avd.natural_sort('name') %}
 #### {{ route_map.name }}
 
-| Sequence | Type | Match and/or Set |
-| -------- | ---- | ---------------- |
+| Sequence | Type | Match and/or Set | Sub-Route-Map | Continue |
```
Comment (jonxstill) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/route-maps.j2]: Are the extra two columns necessary?  There's 150 files changed in this PR, virtually all due to this, with the vast majority of them these two columns just containing a '-'.  The "Match and/or Set" column could become a "Match/Action" column, and you could have continue or sub-route-map actions, with just another couple of if blocks as seen below.

**Review by jonxstill — DISMISSED** [review]
Other than the extra doc columns, looks OK.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/route-maps.j2 (hunk 1 above)
Comment (jonxstill) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/route-maps.j2]: Oops, also didn't realise this was still a draft - apologies!

Comment (emilarista) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/route-maps.j2]: Hi, i guess that could work but updating the column header would still cause the 150 files to change :-) I think it's more clear this way and i do think it's important to document the new knobs in the table.

Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/route-maps.j2]: If we add them in separate columns, we might as well split Match and Set into their own columns as well.

Comment (emilarista) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/route-maps.j2]: I think they should be separate as well. Should i add that in this PR though or separate one?

Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/route-maps.j2]: this one

Comment (emilarista) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/route-maps.j2]: Have split the columns.

**Review by ClausHolbechArista — APPROVED** [review]
LGTM

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/route-maps.j2 [hunk 2]
```diff
@@ -11,6 +11,12 @@ route-map {{ route_map.name }} {{ sequence.type }} {{ sequence.sequence }}
 {%         for match_rule in sequence.match | arista.avd.natural_sort %}
    match {{ match_rule }}
 {%         endfor %}
+{%         if sequence.sub_route_map is arista.avd.defined %}
+   sub-route-map {{ sequence.sub_route_map }}
+{%         endif %}
+{%         if sequence.continue is arista.avd.defined %}
+   continue {{ sequence.continue }}
+{%         endif %}
```
Comment (tgodaA) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/route-maps.j2]: I believe the sequence.continue is not mandatory. This still looks good to me.
Just wondering how do we solve to use only continue without route-map sequence.

image

Probably, we can use the below data-model:

```yaml
continue:
  enabled: < true | false >
  sequence_number: < integer >
```

Comment (tgodaA) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/route-maps.j2]:
```suggestion
{%         if sequence.continue.enabled is arista.avd.defined %}
{%             set continue_cli = "continue" %}
{%             if sequence.continue.sequence_number is arista.avd.defined %}
{%                 set continue_cli = continue_cli ~ " " ~ sequence.continue.sequence_number %}
{%             endif %}
   {{ continue_cli }}
{%         endif %}
```

Comment (emilarista) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/route-maps.j2]: I like the solution, have committed the suggestion.

**Review by ClausHolbechArista — APPROVED** [review]
Still LGTM

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md [hunk 3]
```diff
@@ -644,6 +644,8 @@ route_maps:
           - "< match rule 2 as string >"
         set:
           - "< set as string >"
+        sub_route_map: < sub-route-map name >
+        continue: < sequence number >
```
Comment (tgodaA) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md]:
```suggestion
        continue:
          enabled: < true | false >
          sequence_number: < integer >
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/README_v4.0.md [hunk 4]
```diff
@@ -707,6 +707,8 @@ route_maps:
           - "< match rule 2 as string >"
         set:
           - "< set as string >"
+        sub_route_map: < sub-route-map name >
+        continue: < sequence number >
```
Comment (tgodaA) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/README_v4.0.md]:
```suggestion
        continue:
          enabled: < true | false >
          sequence_number: < integer >
```

**Review by tgodaA — APPROVED** [review]
LGTM

**Pull request merged**